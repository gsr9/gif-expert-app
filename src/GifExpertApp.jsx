import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

function GifExpertApp({ defaultCategories = [] }) {
  //   const categories = ["One punch", "Samurai X", "Dragon Ball"];

  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />

      <AddCategory setCategories={setCategories} />

      <ul>
        {categories.map((c) => (
          <GifGrid key={c} category={c} />
        ))}
      </ul>
    </>
  );
}

export default GifExpertApp;
