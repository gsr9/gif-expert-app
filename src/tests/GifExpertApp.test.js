import { shallow } from "enzyme";
import React from "react";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas en <GifExpertApp />", () => {
  test("debe de mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorÃ­as", () => {
    const categories = ["One punch", "Dragon ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GridGrid").length).toBe(categories.length);
  });
});
