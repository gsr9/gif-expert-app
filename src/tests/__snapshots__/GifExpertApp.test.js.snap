// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul>
    <GifGrid
      category="One punch"
      key="One punch"
    />
  </ul>
</Fragment>
`;
